README.md

```diff
@@ -1,74 +1,129 @@
 
 ## Installation
 
-I recommend using [Pathogen][] for installation. Simply clone
-this repo into your `~/.vim/bundle` directory and you are ready to go.
+### Manual Installation (no plugin manager)
+
+Copy content of this repository into your `~/.vim` directory (or `%HOME%\vimfiles` on Windows).
+
+Be sure that the following lines are in your `.vimrc`
+```vim
+syntax on
+filetype on
+filetype plugin indent on
+```
 
-    cd ~/.vim/bundle
-    git clone https://github.com/purescript-contrib/purescript-vim.git
+### Pathogen
 
-### Manual Installation
+If you are using [Pathogen][], clone this repo into your `~/.vim/bundle` directory and you are ready to go.
 
-Copy content into your `~/.vim` directory (or `%HOME%\vimfiles` on Windows).
+```sh
+cd ~/.vim/bundle
+git clone https://github.com/purescript-contrib/purescript-vim.git
+```
+### vim-plug
 
-Be sure that the following lines are in your
-`.vimrc`
+If you are using [vim-plug][], add the following line in between your `plug#begin` and `plug#end` calls for your vim config file:
 
-    syntax on
-    filetype on
-    filetype plugin indent on
+```vim
+Plug 'purescript-contrib/purescript-vim'
+```
+
+Save and restart (neo)vim and run `:PlugInstall`.
 
 ## Configuration
 
 ### Indentation
 
 To configure indentation in `purescript-vim` you can use the following variables:
 
-* `let purescript_indent_if = 3`
-
-        if bool
-        >>>then ...
-        >>>else ...
+#### g:purescript_indent_case
 
-* `let purescript_indent_case = 5`
+```vim
+let g:purescript_indent_case = 5
+```
 
+```purescript
         case xs of
         >>>>>[]     -> ...
         >>>>>(y:ys) -> ...
+```
 
-* `let purescript_indent_let = 4`
+#### g:purescript_indent_let
 
+```vim
+let g:purescript_indent_let = 4
+```
+
+```purescript
         let x = 0 in
         >>>>x
+```
+
+#### g:purescript_indent_in
+
+```vim
+let g:purescript_indent_in = 1
+```
 
-* `let purescript_indent_where = 6`
+```purescript
+	let x = 0
+	>in x
+```
 
+#### g:purescript_indent_where
+
+```vim
+let g:purescript_indent_where = 6
+```
+
+```purescript
         where f :: Int -> Int
         >>>>>>f x = x
+```
+
+#### g:purescript_indent_do
 
-* `let purescript_indent_do = 3`
+```vim
+let g:purescript_indent_do = 3
+```
 
+```purescript
         do x <- a
         >>>y <- b
+```
 
-* `let purescript_indent_in = 1`
+#### g:purescript_indent_dot
 
-	let x = 0
-	>in x
-
-* `let purescript_indent_dot = v:true`
+```vim
+let g:purescript_indent_dot = 1
+```
 
+```purescript
 	unsnoc
 	  :: forall a
 	  >. List a
 	  -> Maybe (List a, a)
+```
+
+## Contributing
+
+Contributing checklist:
+
+- [ ] Opened an issue before investing a significant amount of work into changes
+- [ ] Update README.md with any new configuration options and behavior
+- [ ] Update CHANGELOG.md with the proposed changes
+- [ ] Run `generate-doc.sh` to re-generate the documentation
```

Comment:
Should any of this be added to an issue or PR template as well? So it's pre-populated there?